pkg/app/pipectl/cmd/installation/controlplane.go

```diff
@@ -0,0 +1,225 @@
+	cmd.Flags().StringVar(&c.cloudSQLServiceAccount, "cloud-sql-service-account-file", c.cloudSQLServiceAccount, "The path to service account which used to access controlplane Google cloud SQL database (if using).")
+	cmd.Flags().StringVar(&c.gcsServiceAccount, "gcs-service-account-file", c.gcsServiceAccount, "The path to service account which used to access controlplane Google Cloud Storage service (if using).")
+	cmd.Flags().StringVar(&c.minioAccessKey, "minio-access-key-file", c.minioAccessKey, "The path to access key which used to connect Minio filestore (if using).")
+	cmd.Flags().StringVar(&c.minioSecretKey, "minio-secret-key-file", c.minioSecretKey, "The path to secret key which used to connect Minio filestore (if using).")
+	cmd.Flags().StringVar(&c.internalTLSKey, "internal-tls-key-file", c.internalTLSKey, "The path to internal TLS key file (if using).")
+	cmd.Flags().StringVar(&c.internalTLSCert, "internal-tls-cert-file", c.internalTLSCert, "The path to internal TLS certificate file (if using).")
+
+	return cmd
+}
+
+func (c *controlplane) run(ctx context.Context, input cli.Input) error {
+	helm, err := c.findHelm()
+	if err != nil {
+		return fmt.Errorf("failed to check required tools before install: %v", err)
+	}
+
+	input.Logger.Info("Installing the controlplane's components...")
+
+	args, err := c.buildHelmArgs()
+	if err != nil {
+		return fmt.Errorf("failed to install controlplane: %v", err)
+	}
+
+	var stderr, stdout bytes.Buffer
+	cmd := exec.CommandContext(ctx, helm, args...)
+	cmd.Stderr = &stderr
+	cmd.Stdout = &stdout
+
+	if err := cmd.Run(); err != nil {
+		return fmt.Errorf("%w: %s", err, stderr.String())
+	}
+
+	input.Logger.Info(stdout.String())
+	input.Logger.Info("Installed the controlplane's components successfully.")
+
+	return nil
+}
+
+func (c *controlplane) findHelm() (string, error) {
+	fi, err := os.Stat(path.Join(c.toolsDir, helmBinaryName))
+	if err != nil && !os.IsNotExist(err) {
+		return "", err
+	}
+
+	// If the Helm executable binary exists in tools dir, use it.
+	if fi != nil {
+		return path.Join(c.toolsDir, helmBinaryName), nil
+	}
+
+	// If the Helm executable binary exists in $PATH, use it.
+	path, err := exec.LookPath("helm")
+	if err != nil {
+		return "", err
+	}
+	return path, nil
+}
+
+func (c *controlplane) buildHelmArgs() ([]string, error) {
+	args := []string{
+		"upgrade",
+		"--install",
+		helmReleaseName,
+		helmChartRepoName,
+		"--version",
+		c.version,
+		"--namespace",
+		c.namespace,
+		"--create-namespace",
+	}
+
+	if c.quickstart {
+		args = append(args,
+			"--values",
+			fmt.Sprintf(helmQuickstartValueRemotePath, c.version),
+		)
+		return args, nil
+	}
+
+	if c.configFile == "" || c.encryptionKeyFile == "" {
+		return nil, fmt.Errorf("missing required fields: config-file or encryption-key-file")
+	}
+
+	args = append(args,
+		"--set-file",
+		fmt.Sprintf("config.data=%s", c.configFile),
```

Comment:
Is there a specific way to specify `--values` in order to override Values.yaml?

Reply:
Nice idea 👍 lets me adopt that.

Reply:
addressed by https://github.com/pipe-cd/pipecd/pull/3835/commits/370e3fc995404c098b06dafe29c8db3395fc2e32 🙏